### PR TITLE
CURATOR-674. Apply code style with spotless

### PR DESCRIPTION
**Pull request by tisonkun**
The first commit is the real plugin and CI change. The second commit is generated by running `./mvnw spotless:apply`.

The discussion thread can be found at https://lists.apache.org/thread/12rwy3y1zcs3pjh9q30851mldgdh4sgg

If we agree on the direction, I'll file a ticket for this.

### CODE REVIEW

**Comment by tisonkun** [comment]
Updated. PTAL @kezhuw @cammckenzie @eolivelli

File: pom.xml [hunk 1]
```diff
@@ -985,6 +992,29 @@
...
                 <version>${maven-dependency-plugin-version}</version>
             </plugin>
+
+            <plugin>
+                <groupId>com.diffplug.spotless</groupId>
+                <artifactId>spotless-maven-plugin</artifactId>
+                <configuration>
+                    <java>
+                        <palantirJavaFormat>
+                            <version>2.30.0</version>
```
Comment (kezhuw) [inline pom.xml:1002]: Add to "properties" tag ?

**Review by kezhuw — APPROVED** [review]
+1

I reviewed the first commit and trust CI for remains.

File: pom.xml (hunk 1 above)
Comment (tisonkun) [reply pom.xml:1002]: Make sense. As this PR already touch a large range of code. I tend to postpone this suggestion after we merge this one.

**Comment by kezhuw** [comment]
I replayed this.
1. git fetch origin
2. gh pr checkout -f 463
3. git reset --hard d8c8bc38c682fa674617907f8fe3112947e8bdf0
4. git rebase origin/master
5. ./mvnw spotless:apply
6. git diff 236d49fc888c6897164419115b69a4bf4f57688d

The final step gives no output and `echo $?` outputs `0`. So I am confident about the remains.

Look forward for this.

**Comment by tisonkun** [comment]
As we agree on the mailing list already, I'll merge this patch in hours.

**Comment by tisonkun** [comment]
Merging...

**Pull request merged**